backends/openvino/tests/ops/base_openvino_op_test.py

```diff
@@ -0,0 +1,167 @@
+
+import numpy as np
+import torch
+from executorch.backends.openvino.partitioner import OpenvinoPartitioner
+from executorch.backends.openvino.preprocess import OpenvinoBackend
+from executorch.exir import EdgeProgramManager, to_edge_transform_and_lower
+from executorch.exir.backend.backend_details import CompileSpec
+from torch.export import export, ExportedProgram
+
+
+class BaseOpenvinoOpTest(unittest.TestCase):
+    device = "CPU"
+    build_folder = ""
+
+    atol = 1e-1
+    rtol = 1e-1
+
+    def execute_layer_test(
+        self,
+        module: torch.nn.Module,
+        sample_inputs: tuple[torch.Tensor],
+        expected_partitions: int = 1,
+        assert_output_equal: bool = True,
+    ):
+
+        module = module.eval()
+        # Export to aten dialect using torch.export
+        aten_dialect: ExportedProgram = export(module, sample_inputs)
+
+        # Convert to edge dialect and lower the module to the backend with a custom partitioner
+        compile_spec = [CompileSpec("device", self.device.encode())]
+        lowered_module: EdgeProgramManager = to_edge_transform_and_lower(
+            aten_dialect,
+            partitioner=[
+                OpenvinoPartitioner(compile_spec),
+            ],
+        )
+
+        # Apply backend-specific passes
+        exec_prog = lowered_module.to_executorch(
+            config=executorch.exir.ExecutorchBackendConfig()
+        )
+
+        # Check if the number of partitions created matches the expected number of partitions
+        self.assertEqual(
+            len(exec_prog.executorch_program.execution_plan[0].delegates),
+            expected_partitions,
+        )
+        # Check if the individual partitions are assigned to Openvino backend
+        for i in range(expected_partitions):
+            self.assertEqual(
+                exec_prog.executorch_program.execution_plan[0].delegates[i].id,
+                OpenvinoBackend.__name__,
+            )
+
+        # Execute the model and compare the outputs with the reference outputs
+        if assert_output_equal:
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                input_list = ""
+                for idx, _ in enumerate(sample_inputs):
+                    input_name = f"input_0_{idx}.raw"
+                    input_list += input_name + " "
+                input_list = input_list.strip() + "\n"
+
+                output_dir = f"{tmp_dir}/outputs"
+
+                # Execute the module in eager mode to calculate the reference outputs
+                ref_output = module(*sample_inputs)
+                if isinstance(ref_output, torch.Tensor):
+                    ref_output = [
+                        ref_output,
+                    ]
+
+                # Serialize the executorch model and save into a temporary file
+                pte_fname = f"{tmp_dir}/openvino_executorch_test.pte"
+                with open(pte_fname, "wb") as file:
+                    exec_prog.write_to_file(file)
```

Comment:
python has temporary directory/file apis. Plus you dont need to write files, actually doing this under test is extremely flaky so I advise against it. ET is load_from_buffer APIs

Reply:
Removed this as tests are using python bindings with the latest updates.